kernel/src/scan/data_skipping.rs

```diff
@@ -304,83 +182,100 @@ impl DataSkippingFilter {
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::*;
+struct DataSkippingPredicateCreator;
+
+impl DataSkippingPredicateEvaluator for DataSkippingPredicateCreator {
+    type Output = Expr;
+    type TypedStat = Expr;
+    type IntStat = Expr;
```

Comment:
impressed how well this API generalizes across data skipping and parquet stats :)